components/eamxx/src/share/field/field.cpp

```diff
@@ -81,6 +84,9 @@ sync_to_host () const {
     default:
       EKAT_ERROR_MSG("Error! Unrecognized field data type in Field::sync_to_host.\n");
   }
+
+  // Return field to read-only state
+  m_is_read_only = true;
```

Comment:
Went for the temporarily disabling read only route. I think this is contained, whereas disabling the checks in `get_view/get_strided_view` could have wider implications.